File: src/main/java/com/josephbaca/entity/LivingEntity.java
package com.josephbaca.entity;

import java.util.ArrayList;
import java.util.List;

/**
 * Any entity that is alive, meaning it can take actions.
 */
public abstract class LivingEntity extends Entity {

  private int health;
  private Inventory inventory = new Inventory();

  LivingEntity(String name, int health, String description) {
    super(name, description);
    this.health = health;
  }

  public int getHealth() {
    return health;
  }

  public void setHealth(int health) {
    this.health = health;
  }

  public abstract int getAttackDamage();

  /**
   * Holds a living entity's {@link Item}.
   */
  class Inventory {

    private List<Item> inventory = new ArrayList<>();

    public String getInventory() {
      return inventory.toString();
    }

    public void add(Item item) {
      inventory.add(item);
    }

  }

}

File: src/main/java/com/josephbaca/entity/LivingEntity.kt
package com.josephbaca.entity

import com.josephbaca.util.Inventory

/**
 * Any entity that is alive, meaning it can take actions.
 */
abstract class LivingEntity protected constructor
    (name: String, health: Int, description: String) : Entity(name, description) {

    var health: Int = health
        set(value) {
            field = if (value < 0) 0 else field
        }
    private val inventory = Inventory()
    abstract val attackDamage: Int

}
